libcxx/include/__assert

```diff
@@ -115,4 +115,10 @@
 #endif // _LIBCPP_HARDENING_MODE == _LIBCPP_HARDENING_MODE_FAST
 // clang-format on
 
+#if _LIBCPP_HARDENING_MODE == _LIBCPP_HARDENING_MODE_NONE
```

Comment:
It would be really nice if these attributes could automatically be added via the `_LIBCPP_ASSERT_VALID_ELEMENT_ACCESS` assertion macro instead, since we pretty much want to have this attribute on each function that has such preconditions.